File: core/src/main/java/lt/markmerkk/app/box2d/temp_components/wheel/WheelImpl.java
package lt.markmerkk.app.box2d.temp_components.wheel;

import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.physics.box2d.World;
import com.badlogic.gdx.physics.box2d.joints.PrismaticJointDef;
import lt.markmerkk.app.box2d.Car;
import org.jetbrains.annotations.NotNull;

public final class WheelImpl extends BaseWheelImpl {
   public void initJoint(
           @NotNull World world,
           @NotNull Car car
   ) {
      PrismaticJointDef jointdef = new PrismaticJointDef();
      jointdef.initialize(car.getBody(), this.getBody(), this.getBody().getWorldCenter(), new Vector2(1.0F, 0.0F));
      jointdef.enableLimit = true;
      jointdef.lowerTranslation = 0.0F;
      jointdef.upperTranslation = 0.0F;
      world.createJoint(jointdef);
   }

   public WheelImpl(
           @NotNull World world,
           @NotNull Car car,
           float posX,
           float posY,
           float width,
           float height
   ) {
      super(world, car, posX, posY, width, height, false);
   }
}

File: core/src/main/java/lt/markmerkk/app/box2d/temp_components/wheel/WheelImpl.kt
package lt.markmerkk.app.box2d.temp_components.wheel

import com.badlogic.gdx.math.Vector2
import com.badlogic.gdx.physics.box2d.*
import com.badlogic.gdx.physics.box2d.joints.PrismaticJointDef
import lt.markmerkk.app.box2d.CarBox2D

/**
 * @author mariusmerkevicius
 * @since 2016-08-28
 */
class WheelImpl(
        world: World,
        carBox2D: CarBox2D,
        posX: Float,
        posY: Float,
        width: Float,
        height: Float
) : BaseWheelImpl(world, carBox2D, posX, posY, width, height, false) {

    override fun initJoint(world: World, carBox2D: CarBox2D) {
        val jointdef = PrismaticJointDef()
        jointdef.initialize(carBox2D.body, body, body.worldCenter, Vector2(1f, 0f))
        jointdef.enableLimit = true
        jointdef.lowerTranslation = 0f
        jointdef.upperTranslation = 0f
        world.createJoint(jointdef)
    }
}